src/main/java/com/microsoftopentechnologies/windowsazurestorage/WAStorageClient.java

```diff
@@ -320,9 +310,12 @@ public static int upload(Run<?, ?> run, Launcher launcher, TaskListener listener
 			// Remove the workspace bit of this path
 			URI srcURI = workspaceURI.relativize(src.toURI());
 
-			InputStream inputStream = src.read();
-			String md5hex = DigestUtils.md5Hex(inputStream);
-			long sizeInBytes = src.length();
+			String md5hex;
+			long sizeInBytes;
+			try (InputStream inputStream = src.read()){
+					md5hex = DigestUtils.md5Hex(inputStream);
+					sizeInBytes = src.length();
```

Comment:
Since the sizeInBytes variable is only used once, suggest just using `src.length()` directly